### ISSUE TITLE
[Enhancement] Allow [CallingConvention] on function pointers defined inline

### ISSUE THREAD

**Issue by blat-blatnik** [beefytech/Beef#1294]
Currently, this doesn't work:

```c#
[CallingConvention(.Stdcall)] // Error: 'CallingConvention' is not valid on this declaration type.
function void(int) foo;
```

So instead I have to do this:

```c#
[CallingConvention(.Stdcall)]
function void Foo(int x);
Foo foo;
```

This works, but it would be nicer to be able to write the first version. This would especially make COM classes look slightly clearer, since every single method in there has to be marked stdcall.

**Comment by blat-blatnik** [comment]
By the way, in corlib System.Windows, none of the COM function pointers are marked stdcall. They still _seem_ to work without destroying the stack, so I'm not sure if that's an actually issue or not - but I thought it'd mention it.

**Comment by bfiete** [comment]
For x86_64 there's no difference in the calling conventions anyway. But, yeah, that would cause an issue for 32-bit compiles.

**Comment by bfiete** [comment]
Fixed at b9908cb4a8b9bd3b40e2594054dd072441ddc4a3

The syntax is `function [CallingConvention(.Stdcall)] void(int) foo;`. This provides a disambiguation between what attributes apply to the member itself and which apply to the function type.
